hack/patch-nix-hash.sh

```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+
+nix --log-format raw build ./dev/nix/
+if [ $? -eq 0 ]; then
+  echo OK
+else
+  # this is reliant on the vendorHash being on line 195
+  sed -i '195s/vendorHash = \"\([^\"]*\)\"/vendorHash = ""/g' ./dev/nix/flake.nix
+  export NIX_HASH=$(nix --log-format raw build ./dev/nix 2>&1 | grep  "got: " | awk '{ split($0,a,"got:    "); print a[2] }')
+  sed -i '195s|vendorHash = \"\([^\"]*\)\"|vendorHash = "'$NIX_HASH'"|g' ./dev/nix/flake.nix
+  echo $NIX_HASH
+fi
```

Comment:
```suggestion
if [ $? -eq 0 ]; then
  echo "No changes to Nix hash needed"
  return
fi

# this is reliant on the vendorHash being on line 195
sed -i '195s/vendorHash = \"\([^\"]*\)\"/vendorHash = ""/g' ./dev/nix/flake.nix
NIX_HASH=$(nix --log-format raw build ./dev/nix 2>&1 | grep  "got: " | awk '{ split($0,a,"got:    "); print a[2] }')
sed -i '195s|vendorHash = \"\([^\"]*\)\"|vendorHash = "'$NIX_HASH'"|g' ./dev/nix/flake.nix
echo "Changed Nix hash to: $NIX_HASH"
```

- can remove the nesting
- can be more explicit in the logging
- I don't think this needs `export`ing? it's only used in the script itself, so it's a local variable, as far as I can tell from this diff

Reply:
Thanks, will change this